ext4/tar2ext4/tar2ext4.go

```diff
@@ -18,6 +18,12 @@ import (
 	"github.com/Microsoft/hcsshim/ext4/internal/format"
 )
 
+const (
+	// the max size in GB of any layer
+	// TODO: upstream with maxLcowVhdSizeGB in containerd 
+	MaxLcowVhdSizeGB = 128 * 1024 * 1024 * 1024
```

Comment:
I don't think we should mention LCOW in this package, even if it's the primary consumer. Is this the max size that the code can currently handle, or some scenario in LCOW can't go over this without failing?

Reply:
If it's just the max the code can handle (and there's nothing we can tweak to get it higher) then `MaxVHDSizeGB` seems like a fine name to me.

Reply:
@dcantah the name is copied from containerd repo https://github.com/containerd/containerd/blob/main/diff/lcow/lcow.go#L47

Reply:
It makes sense to have an lcow nod there, it's for the lcow differ, here I'd leave it off though. But it seems like this is a hard limit

Reply:
agree with @dcantah.
small nits though: can we spell `Max` out? And maybe it's just me, but `MaximumVHDSizeGB` name suggests that we can't go over this number, which isn't really the case here? maybe something like `DefaultMaximumVHDSizeGB` would work better?

Reply:
@dcantah @anmaxvl agree. `DefaultMaximumVHDSizeGB` looks better

Reply:
Well maybe you know the answer judging from the comment @anmaxvl, can the tar2ext4 code deal with a layer larger than 128GB? So this *is* or *isn't* a hard limit?

Reply:
the max max disk size we can handle is 16TB based on: 
https://github.com/microsoft/hcsshim/blob/master/ext4/internal/compactext4/compact.go#L104-L105 and
https://github.com/microsoft/hcsshim/blob/master/ext4/internal/compactext4/compact.go#L1110-L1120.
with this change we're kind of being inconsistent with our internal stuff 🤔

Reply:
so the `defaultMaxDiskSize` is already defined in compact code... I'm afraid this might become confusing, now that I think about it.

Reply:
How about `RecommendedVHDSizeGB`?

Reply:
It doesn't seem like the tar2ext4 package should be exporting a constant which is arbitrary. If this is temporary until the constant is exported from the lcow package in upstream containerd, why not just stick with an internal constant for now?

Reply:
@anmaxvl Ok, I'm remembering we had some bugs in the compact code that hindered the file size, but this was on a file level and not the whole tar I believe (https://github.com/microsoft/hcsshim/pull/958). Maybe I'm missing something for this change. In my mind it was going the opposite way, that Containerd would use this new exported constant and axe the one it has locally in the differ. @praenubilus, could you clarify?

Reply:
> so the `defaultMaxDiskSize` is already defined in compact code... I'm afraid this might become confusing, now that I think about it.

yea..

Reply:
@dcantah I am not familiar with the Containerd repo, need @anmaxvl or @SeanTAllen to give some comments

Reply:
@msscotb , as @dcantah pointed out the goal of this PR is to export a constant that can be later reused here https://github.com/containerd/containerd/blob/main/diff/lcow/lcow.go#L47. 
we could try going containerd route, but we might need to be careful there no to hit any circular dependency issues.

Reply:
Yea, I'd like to get rid of our containerd dep in our root go.mod at some point if possible 😂. But that's good to hear I understood what this was for at least

Reply:
https://github.com/microsoft/hcsshim/pull/1145#discussion_r701503106

I guess we could use that 😄

Reply:
@dcantah , @msscotb any more thoughts on this?

Reply:
after more discussions, I'm leaning towards an internal constant here and name it what it is: `DefaultLcowVHDSizeGB`. Since the only 2 places this is used in hcsshim are:
- https://github.com/microsoft/hcsshim/blob/master/cmd/dmverity-vhd/main.go#L30
- https://github.com/microsoft/hcsshim/blob/master/internal/tools/securitypolicy/main.go#L153
so it seems like the good place for this constant would be in `tar2ext4/dmverity`